src/JitInterface/src/CorInfoImpl.cs

```diff
@@ -1808,7 +1808,14 @@ private void embedGenericHandle(ref CORINFO_RESOLVED_TOKEN pResolvedToken, [Mars
                 // can simply do a static look up
                 pResult.lookup.lookupKind.needsRuntimeLookup = false;
 
-                pResult.lookup.constLookup.handle = (CORINFO_GENERIC_STRUCT_*)ObjectToHandle(_compilation.NodeFactory.NecessaryTypeSymbol(td));
+                if (td.IsArray && !td.IsSzArray)
+                {
+                    // TODO: figure out how JIT can tell us that this call to embedGenericHandle
```

Comment:
Having a tracking issue to clean things up never hurts.

I have created https://github.com/dotnet/coreclr/pull/3341 with the required change.